File: 1_Arrays/2_MaximumPairwiseProduct.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
	int lengthOfVector = nums.size();
	int max_index1 = -1;
	int max_index2 = -1;

	for (int i = 0; i < lengthOfVector; i++)
	{
		if ((max_index1 == -1) || (nums[i] > nums[max_index1]))
		{
			max_index1 = i;
		}
	}

	for (int i = 0; i < lengthOfVector; i++)
	{
		if ((i != max_index1) && ((max_index2 == -1) || (nums[i] > nums[max_index2])))
		{
			max_index2 = i;
		}
	}

        return (nums[max_index1] - 1) * (nums[max_index2] - 1);
        
    }
};

File: 1_Arrays/2_MaximumPairwiseProduct.jl
# Arrays: MaximumPairwiseProduct 
function MaximumPairwiseProduct(nums::Vector{Int})::Int
    maxIndex1 = -1;
    for i = 1 : length(nums)
        if ((maxIndex1 == -1) || (nums[i] > nums[maxIndex1]))
            maxIndex1 = i
        end
    end

    maxIndex2 = -1;
    for i = 1 : length(nums)
        if ((i != maxIndex1) && ((maxIndex2 == -1) || (nums[i] > nums[maxIndex2])))
            maxIndex2 = i
        end
    end
    
    return (nums[maxIndex1]) * (nums[maxIndex2]);
end 


function main()
    nums = [3, 4, 5, 2]
    maximumPairwiseValue = MaximumPairwiseProduct(nums)
    println("MaximumPairwiseProduct = ", maximumPairwiseValue)
end 

main()
